src/plugins/validation/2and3/semantic-validators/schema-ibm.js

```diff
@@ -411,3 +415,29 @@ function isRootSchema(path) {
     current === 'schema' || (parent === 'definitions' && path.length === 2)
   );
 }
+
+// Should produce a warning for properties with the same name but different type
+function checkProperties(schema, propDict, path, messages) {
+  if (has(schema, 'properties')) {
+    const { properties } = schema;
+    const entries = Object.entries(properties);
+    for (const [key, value] of entries) {
+      // skips properties that use $ref
+      if (value.type != undefined) {
+        if (propDict.hasOwnProperty(key)) {
+          if (propDict[key] != value.type) {
+            path += `.properties.${key}`;
+            path = path.replace(/,/g, '.');
+            messages.addMessage(
+              path,
+              `Properties ${key} and ${key} have the same name but different types`,
+              'warning'
```

Comment:
I just realized that I forgot to comment on this - this check should be configurable